File: frontend/src/lib/validations/tileBundleConfig.ts
import { z } from 'zod';

export const tileBundleConfigSchema = z.object({
  enabled: z.boolean().default(true),
  discount: z.number({ required_error: 'Discount is required' }).int().min(0).max(100),
  minLevel: z.number({ required_error: 'Min Level is required' }).int().min(1),
  daysPlayedTrigger: z.number({ required_error: 'Days Played Trigger is required' }).int().min(0),
  sessionsPlayedTrigger: z.number({ required_error: 'Sessions Played Trigger is required' }).int().min(0),
  durationHours: z.number({ required_error: 'Duration Hours is required' }).min(0),
  maxLifetimeShows: z.number({ required_error: 'Max Lifetime Shows is required' }).int().min(1),
  maxSessionShows: z.number({ required_error: 'Max Session Shows is required' }).int().min(1),
  cooldownPopupHours: z.number({ required_error: 'Cooldown Popup Hours is required' }).min(0),
  cooldownOfferHours: z.number({ required_error: 'Cooldown Offer Hours is required' }).min(0),
});

export type TileBundleConfig = z.infer<typeof tileBundleConfigSchema>;

export const defaultTileBundleConfig: TileBundleConfig = {
  enabled: false,
  discount: 0,
  minLevel: 0,
  daysPlayedTrigger: 0,
  sessionsPlayedTrigger: 0,
  durationHours: 0,
  maxLifetimeShows: 0,
  maxSessionShows: 0,
  cooldownPopupHours: 0,
  cooldownOfferHours: 0,
};

File: backend/app/schemas/config_sections/tile_bundle_config.py
from pydantic import BaseModel, Field


class TileBundleConfig(BaseModel):
    """Tile Bundle offer configuration"""
    enabled: bool = Field(default=True, description="Whether the tile bundle offer is enabled")
    discount: int = Field(..., ge=0, le=100, description="Discount percentage (0-100)")
    minLevel: int = Field(..., ge=1, description="Minimum level to show the offer")
    daysPlayedTrigger: int = Field(..., ge=0, description="Days played to trigger the offer")
    sessionsPlayedTrigger: int = Field(..., ge=0, description="Sessions played to trigger the offer")
    durationHours: int = Field(..., ge=1, description="Duration of the offer in hours")
    maxLifetimeShows: int = Field(..., ge=1, description="Maximum lifetime shows")
    maxSessionShows: int = Field(..., ge=1, description="Maximum shows per session")
    cooldownPopupHours: int = Field(..., ge=0, description="Cooldown between popup shows in hours")
    cooldownOfferHours: int = Field(..., ge=0, description="Cooldown between offer shows in hours")
    
    class Config:
        json_schema_extra = {
            "example": {
                "enabled": True,
                "discount": 80,
                "minLevel": 20,
                "daysPlayedTrigger": 2,
                "sessionsPlayedTrigger": 1,
                "durationHours": 24,
                "maxLifetimeShows": 2,
                "maxSessionShows": 1,
                "cooldownPopupHours": 48,
                "cooldownOfferHours": 48
            }
        }







